api/base/settings/defaults.py

```diff
@@ -333,3 +333,5 @@
 }
 
 SLOAN_ID_COOKIE_NAME = 'sloan_id'
+
+MAX_SIZE_OF_ES_QUERY = 10000  # this is a constant for our version of ES.
```

Comment:
So my understanding of this pagination fix is that we're telling Elastic to send us up to 10000 records and then grabbing the appropriate 10 of those records. Is that correct? 

If so, that seems like an inefficient way to do this. Why not have the paginator modify the default query set [with `from` and `size`](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-body.html#request-body-search-from-size)? Or is this being done already somehow?

Reply:
> So my understanding of this pagination fix is that we're telling Elastic to send us up to 10000 records and then grabbing the appropriate 10 of those records. Is that correct?

Yes

> If so, that seems like an inefficient way to do this. Why not have the paginator modify the default query set with from and size? Or is this being done already somehow?

No, it is inefficient but here's my reasoning behind sticking with it.

1.  This is only going to be hit by admins so it's kind of premature optimaization
2. How you design a query in the current view system is pretty much assumed to be independent how you write your query, unless we are fully make an ES ORM, we should stick to common sense queries.

Though I do like the idea of auto pagination built in to django-metrics, effecting the queries here is kind of creeping the scope, but I don't know maybe if it can be done very simply.

Reply:
It _should_ be pretty simple. Play around with it some and see if I'm crazy or not; if it turns out to be a big deal, we can save it for later. The we're going to be using Impact for more and more things as time goes on, so I'd like to set a proper pattern in the beginning if possible.